updated report-editor endpoint new rest api deployment

dknochen opened the pull request:
updated report-editor endpoint to http://secxbrld2.beta.28.io/v1/_queries/public/reports
updated swagger documentation of new report-editor api
newly generated report-api.js

Commit:
updated report-editor endpoint to http://secxbrld.xbrl.io/v1/_queries/public/reports
updated swagger documentation of new report-editor api

wcandillon commented on app/app.js:
```diff
@@ -31,8 +31,8 @@ angular.module('report-editor', [
-            endpoint: 'http://secxbrld.xbrl.io/v1/_queries/public/reports',
-            token: '0ed3b9a9-2795-412d-9863-6186d1cb64bc'
+            endpoint: 'http://secxbrld2.beta.28.io/v1/_queries/public/reports',
+            token: '29143de0-8328-404d-b7f0-591bb871c13f'
```
Create a login page
If a 401 happens, go to login page.
https://github.com/28msec/portal.28.io/blob/master/app/scripts/app.js#L47

wcandillon committed:
Merge branch 'cellfire' into cellfire-d

wcandillon committed:
Fix test

Commit:
Merge branch 'cellfire' of https://github.com/28msec/nolap-report-editor into cellfire-d

Commit:
Merge branch 'cellfire-d' of https://github.com/28msec/nolap-report-editor into cellfire-d

Commit:
ngconstant setup
implemented session class
added session api and report api + wrapper classes
first version of authentication with login page and auth controller

Commit:
removed auto generated files

Commit:
automatically create constants.js
add dir creation

Commit:
Merge branch 'cellfire' of https://github.com/28msec/nolap-report-editor into cellfire-d

Conflicts:
	.gitignore
	app/app.js
	app/modules/report-api.js

Commit:
fixed state conflicts
fixed missing dependencies
fixed session.logout()
fixed ReportAPI usage
moved listReports info reports.js

Commit:
fixed jslint warnings
fixed ignored files

Commit:
Merge branch 'cellfire' of https://github.com/28msec/nolap-report-editor into cellfire-d

Conflicts:
	Gruntfile.js
	app/report/config.js

Commit:
fixed login page
fixed reports controller

Commit:
updated server config api url

wcandillon commented on app/styles/index.less:
```diff
@@ -40,4 +40,20 @@ label {
     &.modal-header-danger {
         background-color: @brand-danger;
     }
+}
```
this should be in app/auth/auth.less
And this file should be included from index.less

wcandillon commented on app/reports/reports.js:
```diff
@@ -129,7 +129,7 @@ angular.module('report-editor')
             report: report.model,
             $method: 'POST',
-            token: ReportEditorConfig.api.token
+            token: $rootScope.session.getToken()
```
This part should use the `API` service.

wcandillon commented on app/auth/auth.js:
```diff
@@ -0,0 +1,34 @@
+            }
+        };
+
+        $scope.back = function() {
```
to be removed

wcandillon commented on app/auth/auth.js:
```diff
@@ -0,0 +1,34 @@
+'use strict';
```
to be cleaned up

wcandillon commented on app/app.js:
```diff
@@ -13,30 +15,34 @@ angular.module('report-editor', [
+            if(HTTP_DEBUG) console.error('HTTP responseError: ' + JSON.stringify(rejection));
+            if(rejection.status === 401) {
+                if(DEBUG) console.error('intercepted 401: emitting auth');
+                $rootScope.$emit('auth');
```
Use `$state.go('login')` instead

wcandillon commented on app/app.js:
```diff
@@ -13,30 +15,34 @@ angular.module('report-editor', [
-    $rootScope.$on('$stateChangeSuccess', function() {
-        ngProgressLite.done();
-    });
+.constant("HTTP_DEBUG", false)
```
to be removed

wcandillon committed:
Code review

Commit:
addressed review comments:
- removed HTTP_DEBUG constant
- integrate API instead of several apis
- some cleanup
